server/src/test/java/org/opensearch/indices/IRCKeyWriteableSerializerTests.java

```diff
@@ -45,6 +45,7 @@ private IndicesRequestCache.Key getRandomIRCKey(int valueLength, Random random,
             value[i] = (byte) (random.nextInt(126 - 32) + 32);
         }
         BytesReference keyValue = new BytesArray(value);
-        return new IndicesRequestCache.Key(shard, keyValue, UUID.randomUUID().toString()); // same UUID source as used in real key
+        return new IndicesRequestCache.Key(shard, keyValue, UUID.randomUUID().toString(), shard.hashCode()); // same UUID
```

Comment:
instead of using `shard.hashCode()` for last parameter, it should be passed in from the caller of `getRandomIRCKey` using `System.identityHashCode(indexShard)`

Reply:
I will take care of this in separate PR.

Reply:
@sohami I had addressed this here - https://github.com/opensearch-project/OpenSearch/pull/13597